src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -215,6 +216,199 @@ public bool Read()
             return retVal;
         }
 
+        /// <summary>
+        /// Compares the UTF-8 encoded text to the unescaped JSON token value in the source and returns true if they match.
+        /// </summary>
+        /// <param name="otherUtf8Text">The UTF-8 encoded text to compare against.</param>
+        /// <returns>True if the JSON token value in the source matches the UTF-8 encoded look up text.</returns>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if trying to find a text match on a JSON token that is not a string
+        /// (i.e. other than <see cref="JsonTokenType.String"/> or <see cref="JsonTokenType.PropertyName"/>).
+        /// <seealso cref="TokenType" />
+        /// </exception>
+        /// <remarks>
+        /// If the look up text is invalid UTF-8 text, the method will always return false since you cannot have 
+        /// invalid UTF-8 within the JSON payload.
+        /// </remarks>
+        /// <remarks>
+        /// The comparison of the JSON token value in the source and the look up text is done by first unescaping the JSON value in source,
+        /// if required. The look up text is matched as is, without any modifications to it.
+        /// </remarks>
+        public bool TextEquals(ReadOnlySpan<byte> otherUtf8Text)
```

Comment:
Should `otherUtf8Text` contain `unescaped` in the name?  Or is that meant to be conveyed by `text`?

Reply:
I couldn't come up with a good name that encompasses all these facts:
- has to be utf8
- is text data
- It's the value we are comparing to
- It's unescaped

Also, the input text to compare against doesn't necessarily have to be unescaped. In fact, it probably should be **escaped**.

Any suggestions? Drop the `other`? `utf8EscapedText`?